### PR TITLE
[Review] Make cmake generate c++ tests inside gtests folder

### PR DESCRIPTION
**Pull request by aucahuasi**
Also this PR improve the documentation based on the feedback of the PR https://github.com/gpuopenanalytics/libgdf/pull/60

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,8 +98,6 @@ if(HASH_JOIN)
   set(CUDA_NVCC_FLAGS  ${CUDA_NVCC_FLAGS};-DHASH_JOIN)
 endif()
 
-find_package(CUDA)
-
```
Comment (scopatz) [inline CMakeLists.txt]: This doesn't seem correct to me.

Comment (aucahuasi) [reply CMakeLists.txt]: Is duplicated, see lines above

Comment (aucahuasi) [reply CMakeLists.txt]: That line was repeated without reason that is why i removed

Comment (scopatz) [reply CMakeLists.txt]: Oh yes, I see now. Sorry for the noise.

Comment (aucahuasi) [reply CMakeLists.txt]: No problem, thanks for taking the time to review this stuff ;)

**Pull request merged**